### PR TITLE
[minor] Support management of CIS in IBM Cloud

### PR DESCRIPTION
**Pull request by Ashish-Mahawal1**
Issue :  https://github.ibm.com/wiotp/tracker/issues/10620

Changes :
1. Added new role `cis` to mas_devops

### Provisioning flow
```
a. Provision CIS Instance in customer account
b. Add customer domain to customer's CIS Instance
c. Configure Domain settings in customer CIS Instance
d. Add DNS Records of type `NS` for customer's Domain nameservers to Master CIS Account
```

### Deprovisioning Flow
```
a. Delete DNS Record from Master Account
b. Delete Domain from Customer Account
c. Delete Customer CIS Instance
```

Flow Diagram 

CIS Role updated

### Testing
Manual Tests
1. Tested provisioning flow when no resources already exists 
2. Tested deprovisioing flow when all CIS related resources exists
3. Tested provisioning flow when all resources already exists
4. Tested deprovisioing flow when some CIS related resources does not exists

### CODE REVIEW

File: ibm/mas_devops/roles/cis/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+#vars for ibmcloud
+ibmcloud_apikey: "{{ lookup('env', 'IBMCLOUD_APIKEY') }}"
+master_ibmcloud_api_key: "{{ lookup('env', 'MASTER_IBMCLOUD_APIKEY') }}"
+ibmcloud_region: "{{ lookup('env', 'IBMCLOUD_REGION') | default('us-east', True) }}"
+master_cis_resource_group: "{{ lookup('env', 'MASTER_CIS_RESOURCE_GROUP') | default('manager', True) }}"
+ibmcloud_resourcegroup: "{{ lookup('env', 'IBMCLOUD_RESOURCEGROUP') | default('Default', true) }}"
+master_cis_resource_name: "{{ lookup('env', 'MASTER_CIS_RESOURCE_NAME') | default('masms-cis', True) }}"
```
Comment (Ashish-Mahawal1) [inline ibm/mas_devops/roles/cis/defaults/main.yml:8]: change  default value to <mas-instance-id>-cis

File: ibm/mas_devops/roles/cis/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+#vars for CIS provision
+cis_action: "{{ lookup('env', 'CIS_ACTION') | default('provision', True) }}"
+masms_base_domain: "{{ lookup('env', 'MASMS_BASE_DOMAIN') | default('suite.maximo.com', True) }}"
```
Comment (Ashish-Mahawal1) [inline ibm/mas_devops/roles/cis/defaults/main.yml:11]: rename var to master_cis_...
remove default value

File: ibm/mas_devops/roles/cis/meta/main.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+galaxy_info:
+  author: Ashish Mahawal
+  description: Configure ceph object storage or ibm cloud object storage for MAS Assist
```
Comment (durera) [inline ibm/mas_devops/roles/cis/meta/main.yml:3]: This should be corrected

File: docs/changes.md [hunk 4]
```diff
@@ -1,6 +1,7 @@
 ## Changes
 Note that links to pull requests prior to public release of the code (4.0) direct to IBM GitHub Enterprise, and will only be accessible to IBM employees.
 
+- `12.6` Support IBM CIS provisioning deprovisioning ([#549](https://github.com/ibm-mas/ansible-devops/pull/549/))
```
Comment (durera) [inline docs/changes.md:4]: Merge from master and update the other changelog too

File: build/bin/copy-role-docs.sh [hunk 5]
```diff
@@ -71,3 +71,4 @@ copyDoc suite_mustgather_download
 copyDoc suite_upgrade
 copyDoc suite_verify
 copyDoc uds
+copyDoc cis
```
Comment (durera) [inline build/bin/copy-role-docs.sh:74]: add this in alphabetical order like the rest please.  Also add this to appropriate section in mkdocs.yml

**Review by durera — CHANGES_REQUESTED** [review]

File: ibm/mas_devops/roles/cis/meta/main.yml (hunk 3 above)
Comment (Ashish-Mahawal1) [reply ibm/mas_devops/roles/cis/meta/main.yml]: done

**Pull request merged**